runtime/dart_isolate_unittests.cc

```diff
@@ -592,6 +592,101 @@ TEST_F(DartIsolateTest, DartPluginRegistrantIsCalled) {
   ASSERT_EQ(messages[0], "_PluginRegistrant.register() was called");
 }
 
+TEST_F(DartIsolateTest, SpawningAnIsolateDoesNotReloadKernel) {
+  ASSERT_FALSE(DartVMRef::IsInstanceRunning());
+  auto settings = CreateSettingsForFixture();
+  auto vm_ref = DartVMRef::Create(settings);
+  ASSERT_TRUE(vm_ref);
+  auto vm_data = vm_ref.GetVMData();
+  ASSERT_TRUE(vm_data);
+  TaskRunners task_runners(GetCurrentTestName(),    //
+                           GetCurrentTaskRunner(),  //
+                           GetCurrentTaskRunner(),  //
+                           GetCurrentTaskRunner(),  //
+                           GetCurrentTaskRunner()   //
+  );
+
+  ASSERT_TRUE(settings.application_kernels);
+  auto mappings = settings.application_kernels();
+  ASSERT_EQ(mappings.size(), 1u);
+
+  size_t get_kernel_count = 0u;
+  // This feels a little brittle, but the alternative seems to be making
+  // DartIsolate have virtual methods so it can be mocked or exposing weird
+  // test-only API on IsolateConfiguration.
+  settings.application_kernels =
+      [&get_kernel_count, mapping = mappings.front().release()]() -> Mappings {
+    get_kernel_count++;
+    std::vector<std::unique_ptr<const fml::Mapping>> kernel_mappings;
+    kernel_mappings.emplace_back(std::unique_ptr<const fml::Mapping>(mapping));
```

Comment:
This will crash if the test ever fails since it will double release mapping.  I'd just copy the mapping instead.

Reply:
I can improve the failure mode, but mappings aren't copiable.

Reply:
You can copy them.  You just make a MallocMapping and do a `memcpy`.

Reply:
Resolving because of dan's note above.  We can make this not crash by just failing earlier.